mgmtsystem/openstack.py

```diff
@@ -215,6 +215,7 @@ def delete_vm(self, instance_name):
             if floating_ips:
                 floating_ip = floating_ips[0]
         instance.delete()
+        wait_for(lambda: not self.does_vm_exist(instance_name), timeout='3m', delay=5)
```

Comment:
I wonder if it should actually be the other way around? We disassociate the floating IP BEFORE we delete the instance. The complaints in the log is about a port being missing. This would probably happen if we removed the instance first. What say you?